pi-geofence/src/main/java/com/ibm/pi/geofence/GeofencingJSONUtils.java

```diff
@@ -0,0 +1,283 @@
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Utility methods to parse one or more geofences in geojson format.
+ */
+class GeofencingJSONUtils {
+    /**
+     * Logger for this class.
+     */
+    private static final Logger log = LoggingConfiguration.getLogger(PIGeofencingManager.class.getSimpleName());
+    /**
+     * Date format used to convert dates from/to UTC format such as "2015-08-24T09:00:00-05:00".
+     */
+    static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    /**
+     * Date format used to convert dates from/to UTC format such as "2015-08-24T09:00:00.123Z".
+     */
+    private static final String TMP_DESC = UUID.randomUUID().toString();
+
+    /**
+     * Parse a list of geofences.
+     * @param json json object representing the list of fences.
+     * @return a list of {@link PIGeofence} instances.
+     * @throws Exception if a parsing error occurs.
+     */
+    static GeofenceList parseGeofences(JSONObject json) throws Exception {
+        List<PersistentGeofence> result = new ArrayList<>();
+        JSONArray features = json.getJSONArray("features");
+        for (int i=0; i<features.length(); i++) {
+            JSONObject feature = features.getJSONObject(i);
+            result.add(parseGeofence(feature));
+        }
+        if (json.has("properties")) {
+            JSONObject properties = json.getJSONObject("properties");
+            List<String> deletedCodes = null;
+            if (properties.has("deleted")) {
+                JSONArray deletedCodesJson = properties.getJSONArray("deleted");
+                deletedCodes = new ArrayList<>(deletedCodesJson.length());
+                for (int i=0; i<deletedCodesJson.length(); i++) {
+                    deletedCodes.add(deletedCodesJson.getString(i));
+                }
+                if (!deletedCodes.isEmpty()) {
+                    int n = GeofencingUtils.deleteGeofences(deletedCodes);
+                    log.debug(String.format("deleted %d geofences from local DB", n));
+                }
+            }
+            int totalGeofences = properties.has("totalFeatures") ? properties.getInt("totalFeatures") : -1;
+            String lastSyncDate = properties.has("lastSyncDate") ? properties.getString("lastSyncDate") : null;
+            return new GeofenceList(result, totalGeofences, lastSyncDate, deletedCodes);
+        }
+        return new GeofenceList(result);
+    }
+
+    /**
+     * Parse a single geofence.
+     * @param feature json object representing the fence.
+     * @return a {@link PIGeofence} instance.
+     * @throws Exception if a parsing error occurs.
+     */
+    static PersistentGeofence parseGeofence(JSONObject feature) throws Exception {
+        JSONObject props = feature.getJSONObject("properties");
+        String code = props.has("code") ? props.getString("code") : (props.has("@code") ? props.getString("@code") : null);
+        JSONObject updatedJSON = props.getJSONObject("@updated");
```

Comment:
Are you not checking if props has @updated or @created since these values are required by any document in Cloudant? It would make sense, just checking.